tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -899,6 +912,12 @@ void CommandQueue::enqueue_read_buffer(Buffer& buffer, void* dst, bool blocking)
         tt::log_debug(tt::LogDispatch, "EnqueueReadBuffer");
 
         uint32_t available_space_bytes = DeviceCommand::HUGE_PAGE_SIZE - (get_cq_completion_wr_ptr(this->device->id()) << 4);
+        if (available_space_bytes < padded_page_size) {
+            // wrap the completion region because a single page won't fit in available space
+            this->wrap(DeviceCommand::WrapRegion::COMPLETION);
```

Comment:
@DrJessop in this commit I resolve a hang that I was hitting because space left in completion region was less than a page. In that case host issues a wrap to completion region. 

I updated wrap index to accept one of: 
```
    enum class WrapRegion : uint8_t {
        NONE = 0,
        ISSUE = 1,
        COMPLETION = 2
    };
// defined in device_command.hpp
```

Reply:
Shouldn't this be a function of the number of pages that you're trying to read in one shot into the completion queue (rather than just 1 padded page size)?

Reply:
ah yeah so i was using the available size to figure out how big of a read to issue but I can update the logic to try and read in one shot and only issue multiple reads if buffer size is larger than completion q size